ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py

```diff
@@ -187,3 +187,57 @@ def test_torch_cumsum(
         dtype=input_dtype,
         out=None,
     )
+
+
+@given(
+    dtype_and_values=helpers.dtype_and_values(
+        available_dtypes=tuple(
+            set(ivy_np.valid_float_dtypes).intersection(
+                set(ivy_torch.valid_float_dtypes)
+            ),
+        ),
+        shape=helpers.get_shape(min_num_dims=2),
+    ),
+    offset=helpers.get_axis(
+        shape=st.shared(helpers.get_shape(min_num_dims=1), key="shape"),
+        force_int=True
+    ),
+    dim1=helpers.get_axis(
+        shape=st.shared(helpers.get_shape(min_num_dims=1), key="shape"),
+        force_int=True
+    ),
+    dim2=helpers.get_axis(
+        shape=st.shared(helpers.get_shape(min_num_dims=1), key="shape"),
+        force_int=True
+    ),
+    as_variable=st.booleans(),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.torch.diagonal"
+    ),
+    native_array=st.booleans(),
+)
+def test_torch_diagonal(
+        dtype_and_values,
+        offset,
+        dim1,
+        dim2,
+        as_variable,
+        num_positional_args,
+        native_array,
+        fw,
+):
+    input_dtype, value = dtype_and_values
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=True,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="torch",
+        fn_tree="diagonal",
+        input=np.asarray(value, dtype=input_dtype),
+        offset=offset,
+        axis1=dim1,

```

Comment:
the keyword parameters here should match the function signature for `diagonal`, so it should be `dim1=dim1` and `dim2=dim2`